### ISSUE TITLE
Modal windows and click-through

### ISSUE THREAD

**Issue by gajop** [Spring-Chobby/Chobby#12]
Sometimes it's possible to click through chili windows to select components underneath. This is particularly annoying when the chili theme isn't transparent.
Modal windows should also enforce that no other window/component can be clicked while the current one is being interacted with, essentially like dialogues should work.

**Comment by gajop** [comment]
The Priority Popup implementation is something that we'll probably end up using throughout the code, but I really want to have native Chili support for this.
Modal windows (https://en.wikipedia.org/wiki/Modal_window) are a well known GUI concept and should be supported by Chili itself.

I suggest that all Chili windows with "modal = true" act like modal windows (preventing any interaction until they're disposed).

**Comment by GoogleFrog** [comment]
I feel like our design works around this issue sufficiently to call it solved.

**Comment by gajop** [comment]
Yes, it works, but I want it added to Chili so other projects can benefit (e.g. I plan to use it in Scened as well).
Modal windows are a well known GUI concept that they have their wiki article, so it makes sense to do it "natively" in Chili.
